src/routes/GettingStartedPage/Wizard/InstallOptions.tsx

```diff
@@ -24,12 +27,35 @@ const GKE = 'gke';
 const OPERATOR = 'operator';
 const HELM = 'helm';
 
-const InstallOptions = ({
+const environments = [OPENSHIFT, EKS, AKS, GKE, OPERATOR, HELM] as const;
+
+export type Environment = (typeof environments)[number];
+
+function isEnvironment(e: string): e is Environment {
+  return environments.some((env) => e === env);
+}
+
+const installationMethods = [OPERATOR, HELM] as const;
+
+export type InstallationMethod = (typeof installationMethods)[number];
```

Comment:
could you explain the right side of this statement for me? i'm having a hard time remembering what the `(...)[number]` means 😅

Reply:
Sure, the `(typeof installationMethods)` evaluates to a tuple that is exactly identical to the `const installationMethods = [OPERATOR, HELM] as const;` above. (The `as const` qualifier at the end ensures that this only allows `'operator'` or `'helm'`, instead of `string`.

Then the `[number]` piece causes TypeScript to evaluate the type as a union of any item in the tuple, and is equivalent to `export type InstallationMethod = 'operator' | 'helm';`.

The pattern is useful when we have a static list of primitives at runtime, and we want to create a type that represents any single one of those primitives.

This is usually combined with the third item below: `isInstallation`, which gives us a function at runtime to ensure a more general value conforms to the type union above.